core/commands/repository/interactors/fetch_repository.py

```diff
@@ -1,14 +1,27 @@
+from shared.django_apps.codecov_auth.models import Owner
+from shared.django_apps.core.models import Repository
+
 from codecov.commands.base import BaseInteractor
 from codecov.db import sync_to_async
-from core.models import Repository
 
 
 class FetchRepositoryInteractor(BaseInteractor):
     @sync_to_async
-    def execute(self, owner, name):
+    def execute(
+        self,
+        owner: Owner,
+        name: str,
+        okta_authenticated_accounts: list[int],
+        exclude_okta_enforced_repos: bool = True,
+    ):
+        queryset = Repository.objects.viewable_repos(self.current_owner)
```

Comment:
I'm curious if it would make sense for viewable_repos itself to have this kind of check. This works for now, I'm wondering though if we're going to have to exclude these on a "resolver to resolver" basis you know?

Reply:
Having the `viewable_repos` not exclude okta is pretty convenient for being more specific about how to filter repos. For instance, the `RepositoryViewSetMixin` calls `viewable_repos` but without `exclude_accounts_enforced_okta`. The only reason why I have this conditional here is because this interactor is shared with the legacy upload API. Otherwise, if this interactor is only for GraphQL, then we can just have `exclude_accounts_enforced_okta` by default.